src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -617,7 +617,7 @@ public ParseResult ParseAndBind(ScriptFile scriptFile, ConnectionInfo connInfo)
                 AutoCompleteHelper.PrepopulateCommonMetadata(info, scriptInfo, this.BindingQueue);
 
                 // Send a notification to signal that autocomplete is ready
-                ServiceHost.Instance.SendEvent(IntelliSenseReadyNotification.Type, new IntelliSenseReadyParams() {OwnerUri = info.OwnerUri});
+                TelemetryHelper.SendEventAndIgnoreErrors(IntelliSenseReadyNotification.Type, new IntelliSenseReadyParams() {OwnerUri = info.OwnerUri});
```

Comment:
oops, I thought this fix was about telemetry..this looks like a product notification..sorry for the confusion.

In this scenario, I think we should continue to log through ServiceHost but have some type of "test flag" to prevent throwing as you were suggesting.